android/java/org/chromium/chrome/browser/ntp/IncognitoNewTabPageView.java

```diff
@@ -0,0 +1,229 @@
+
+        View learnMore = findViewById(R.id.learn_more);
+        learnMore.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                mManager.loadIncognitoLearnMore();
+            }
+        });
+
+        mWidthDp = getContext().getResources().getConfiguration().screenWidthDp;
+        mHeightDp = getContext().getResources().getConfiguration().screenHeightDp;
+
+        adjustView();
+    }
+
+    /**
+     * Initialize the incognito New Tab Page.
+     * @param manager The manager that handles external dependencies of the view.
+     */
+    void initialize(IncognitoNewTabPageManager manager) {
+        mManager = manager;
+    }
+
+    /** @return The IncognitoNewTabPageManager associated with this IncognitoNewTabPageView. */
+    protected IncognitoNewTabPageManager getManager() {
+        return mManager;
+    }
+
+    /**
+     * @see org.chromium.chrome.browser.compositor.layouts.content.
+     *         InvalidationAwareThumbnailProvider#shouldCaptureThumbnail()
+     */
+    boolean shouldCaptureThumbnail() {
+        if (getWidth() == 0 || getHeight() == 0) return false;
+
+        return getWidth() != mSnapshotWidth
+                || getHeight() != mSnapshotHeight
+                || mScrollView.getScrollY() != mSnapshotScrollY;
+    }
+
+    /**
+     * @see org.chromium.chrome.browser.compositor.layouts.content.
+     *         InvalidationAwareThumbnailProvider#captureThumbnail(Canvas)
+     */
+    void captureThumbnail(Canvas canvas) {
+        ViewUtils.captureBitmap(this, canvas);
+        mSnapshotWidth = getWidth();
+        mSnapshotHeight = getHeight();
+        mSnapshotScrollY = mScrollView.getScrollY();
+    }
+
+    @Override
+    public void onAttachedToWindow() {
+        super.onAttachedToWindow();
+        assert mManager != null;
+        if (mFirstShow) {
+            mManager.onLoadingComplete();
+            mFirstShow = false;
+        }
+    }
+
+    @Override
+    public void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        // View#onConfigurationChanged() doesn't get called when resizing this view in
+        // multi-window mode, so #onMeasure() is used instead.
+        Configuration config = getContext().getResources().getConfiguration();
+        if (mWidthDp != config.screenWidthDp || mHeightDp != config.screenHeightDp) {
+            mWidthDp = config.screenWidthDp;
+            mHeightDp = config.screenHeightDp;
+            adjustView();
+        }
+
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+    }
+
+    private void adjustView() {
+        adjustIcon();
+        adjustLayout();
+        adjustLearnMore();
+    }
+
+    /** Adjusts the paddings, margins, and the orientation of bulletpoints. */
+    private void adjustLayout() {
+        int paddingHorizontalDp;
+        int paddingVerticalDp;
+
+        boolean bulletpointsArrangedHorizontally;
```

Comment:
Nit: This variable doesn't seem to be used for anything.

Reply:
Actually, does this function serve any purpose at all? Seems like it just initializes some local variables and then exits.

Reply:
removed